example/newreposecret/main.go

```diff
@@ -0,0 +1,172 @@
+		log.Fatal(err)
+	}
+
+	fmt.Printf("Added secret %q to the repo %v/%v\n", secretName, *owner, *repo)
+}
+
+func getSecretName() (string, error) {
+	secretName := flag.Arg(0)
+	if secretName == "" {
+		return "", fmt.Errorf("missing argument secret name")
+	}
+	return secretName, nil
+}
+
+func getSecretValue(secretName string) (string, error) {
+	secretValue := os.Getenv(secretName)
+	if secretValue == "" {
+		return "", fmt.Errorf("secret value not found under env variable %q", secretName)
+	}
+	return secretValue, nil
+}
+
+// githubAuth returns a GitHub client and context.
+func githubAuth(token string) (context.Context, *github.Client, error) {
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	client := github.NewClient(tc)
+	return ctx, client, nil
+}
+
+// addRepoSecret will add a secret to a GitHub repo for use in GitHub Actions.
+//
+// Finally the secretName and secretValue will determine the name of the secret added and it's corresponding value.
+//
+// The actual transmission of the secret value to GitHub using the api requires that the secret value is encrypted
+// using the public key of the target repo. This encryption must be done using sodium. This function has a hard
+// dependency on sodium being installed where this is run. Sodium can be installed from this url:
+// https://formulae.brew.sh/formula/libsodium
+//
+// First step of the upload process by this function is to get the public key of the repo. The public key comes base64
+// encoded, so it must be decoded prior to use in sodiumlib.
+//
+// Second the secret value starts as a string type, but must be converted into a slice of bytes.
+// Third the decoded public key of the repo is used to encrypt the secret with sodium.CryptoBoxSeal resulting
+// in a secret encrypted as bytes
+//
+// Finally the secret bytes need to be encoded as a base64 string and used in a github.EncodedSecret type
+// that is passed into the GitHub client.Actions.CreateOrUpdateRepoSecret method to populate the secret in GitHub.
+//
+// The other two properties of the github.EncodedSecret type are the name of the secret to be added (string not base64)
+// and the KeyID of the public key used to encrypt the secret, which is gettable from the public key's GetKeyID method.
+//
+// Finally it passes in the github.EncodedSecret object to CreateOrUpdateRepoSecret which creates or updates the secret
+// in GitHub
+func addRepoSecret(ctx context.Context, client *github.Client, owner string, repo, secretName string, secretValue string) error {
+	publicKey, _, err := client.Actions.GetRepoPublicKey(ctx, owner, repo)
+	if err != nil {
+		return err
+	}
+
+	encryptedSecret, err := encryptSecretWithPublicKey(publicKey, secretName, secretValue)
+	if err != nil {
+		return err
+	}
+
+	_, err = client.Actions.CreateOrUpdateRepoSecret(ctx, owner, repo, encryptedSecret)
```

Comment:
I'm going to recommend this again to keep the code a bit more concise.
You typically do this when there is not another value that you are keeping around for other use.

```suggestion
	if _, err := client.Actions.CreateOrUpdateRepoSecret(ctx, owner, repo, encryptedSecret); err != nil {
```